Source/Scene/GroundPrimitiveBatch.js

```diff
@@ -0,0 +1,997 @@
+            command.shaderProgram = primitive._sp;
+            command.uniformMap = uniformMap;
+            command.boundingVolume = bv;
+            command.pass = Pass.GROUND;
+        }
+    }
+
+    function createPickCommands(primitive) {
+        if (!primitive._pickCommandsDirty) {
+            return;
+        }
+
+        var length = primitive._indexOffsets.length * 3;
+        var pickCommands = primitive._pickCommands;
+        pickCommands.length = length;
+
+        var vertexArray = primitive._va;
+        var uniformMap = primitive._batchTable.getPickUniformMapCallback()(primitive._uniformMap);
+
+        for (var j = 0; j < length; j += 3) {
+            var offset = primitive._indexOffsets[j / 3];
+            var count = primitive._indexCounts[j / 3];
+            var bv = primitive._boundingVolumes[j / 3];
+
+            // stencil preload command
+            var command = pickCommands[j];
+            if (!defined(command)) {
+                command = pickCommands[j] = new DrawCommand({
+                    owner : primitive
+                });
+            }
+
+            command.vertexArray = vertexArray;
+            command.offset = offset;
+            command.count = count;
+            command.renderState = primitive._rsStencilPreloadPass;
+            command.shaderProgram = primitive._sp;
+            command.uniformMap = uniformMap;
+            command.boundingVolume = bv;
+            command.pass = Pass.GROUND;
+
+            // stencil depth command
+            command = pickCommands[j + 1];
+            if (!defined(command)) {
+                command = pickCommands[j + 1] = new DrawCommand({
+                    owner : primitive
+                });
+            }
+
+            command.vertexArray = vertexArray;
+            command.offset = offset;
+            command.count = count;
+            command.renderState = primitive._rsStencilDepthPass;
+            command.shaderProgram = primitive._sp;
+            command.uniformMap = uniformMap;
+            command.boundingVolume = bv;
+            command.pass = Pass.GROUND;
+
+            // color command
+            command = pickCommands[j + 2];
+            if (!defined(command)) {
+                command = pickCommands[j + 2] = new DrawCommand({
+                    owner : primitive
+                });
+            }
+
+            command.vertexArray = vertexArray;
+            command.offset = offset;
+            command.count = count;
+            command.renderState = primitive._rsPickPass;
+            command.shaderProgram = primitive._spPick;
+            command.uniformMap = uniformMap;
+            command.boundingVolume = bv;
+            command.pass = Pass.GROUND;
+        }
+
+        primitive._pickCommandsDirty = false;
+    }
+
+    /**
+     * Colors the entire tile when enabled is true. The resulting color will be (polygon batch table color * color).
+     * @private
```

Comment:
Since the class is marked private, there's no need to mark everything else private.